### DISCUSSION TITLE
Can I validate a user using a OIDC token emitted by a known OIDC provider?

### DISCUSSION THREAD

**Discussion by juan131** [ory/kratos#2430] [Q&A] [answered]
According to the documentation, Kratos doesn't store [OIDC tokens in the database](https://www.ory.sh/docs/kratos/guides/sign-in-with-github-google-facebook-linkedin#how-ory-kratos-handles-oidc-token) once the OIDC provider sends it back, instead it gets the user information and returns a session token.

Assuming that **I have a OIDC token that was obtained outside Kratos context**, could I use it to obtain a session object (e.g. making a call to `/sessions/whoami` endpoint or similar)?

**Comment by aeneasr** [comment, marked as answer]
Not yet, but there is a PR for this: https://github.com/ory/kratos/pull/2346

**Reply by vinckr** [reply]
Hey @juan131 ,
Just checking in if your question was answered or if there was still something unclear.
I think we will have some progress in that PR very soon!
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
